core/services/workflow_service/controllers/workflow_controller.py

```diff
@@ -0,0 +1,340 @@
+
+    # Convert to Airflow-compatible representation
+    for node, data in graph.nodes(data=True):
+        task_id = _cb_id_to_task_id(node)
+        parts.append(templates["algorithm"].render(
+            task_id=task_id,
+            image=data["image"],
+            name=data["name"],
+            uuid=data["uuid"],
+            entry_name=data["entry_name"],
+            project=str(project_uuid),
+            environment=data["environment"],
+            local_storage_path_external="/tmp/scystream-data",
+        ))
+
+    # Render dependencies
+    parts.extend([
+        templates["dependency"].render(
+            from_task=_cb_id_to_task_id(from_task),
+            to_task=_cb_id_to_task_id(to_task)
+        )
+        for from_task, to_task in graph.edges
+    ])
+
+    return "\n".join(parts)
+
+
+def save_dag_to_file(dag_code, dag_id):
+    os.makedirs(DAG_DIRECTORY, exist_ok=True)
+    filename = os.path.join(DAG_DIRECTORY, f"{dag_id}.py")
+
+    with open(filename, "w") as f:
+        f.write(dag_code)
+
+    return filename
+
+
+def validate_value(value: str) -> bool:
+    return value is None or value == "" or value == []
+
+
+def validate_workflow(project_uuid: UUID):
+    """
+    Checks:
+        - Are there compute blocks?
+        - Are all envs and configs set?
+    """
+    blocks_in_project = compute_block_controller.get_compute_blocks_by_project(
+        project_uuid
+    )
+
+    if len(blocks_in_project) == 0:
+        raise HTTPException(
+            status_code=422, detail="Project is missing blocks.")
+
+    # Collect all inputs & outputs
+    confs = {}
+    missing_values = {}
+
+    for block in blocks_in_project:
+        block_id = block.uuid
+
+        for ek, ev in block.selected_entrypoint.envs.items():
+            confs[ek] = ev
+            if validate_value(ev):
+                missing_values.setdefault(str(block_id), []).append(ek)
+
+        for io in block.selected_entrypoint.input_outputs:
+            for ck, cv in io.config.items():
+                confs[ck] = cv
+                if validate_value(cv):
+                    missing_values.setdefault(str(block_id), []).append(ck)
+
+    if missing_values:
+        raise HTTPException(
+            status_code=422, detail=WorfklowValidationError(
+                project_id=str(project_uuid),
+                missing_configs=missing_values
+            ).dict()
+        )
+
+
+def translate_project_to_dag(project_uuid: UUID) -> str:
+    """
+    Parses a project and its blocks into a DAG, validates it, and saves it.
+    """
+    project = read_project(project_uuid)  # Ensures project is set
+    graph = create_graph(project)
+    templates = init_templates()
+    dag_id = _project_id_to_dag_id(project_uuid)
+    dag_code = gen_dag_code(graph, templates, dag_id, project_uuid)
+    save_dag_to_file(dag_code, dag_id)
+
+    # Make sure airflow has enough time to create the dag internally
+    time.sleep(1)
```

Comment:
Actually I dont really feel that logic.
We should implement an iterativ checking. Not every airflow instance is set like the version we introduced in `docker-compose.yml`.